rs/types/management_canister_types/src/lib.rs

```diff
@@ -773,37 +773,51 @@ impl UninstallCodeArgs {
 
 impl Payload<'_> for UninstallCodeArgs {}
 
+/// Maximum number of allowed log viewers (specified in the interface spec).
+const MAX_ALLOWED_LOG_VIEWERS_COUNT: usize = 10;
+
+pub type BoundedAllowedViewers =
+    BoundedVec<MAX_ALLOWED_LOG_VIEWERS_COUNT, UNBOUNDED, UNBOUNDED, PrincipalId>;
+
 /// Log visibility for a canister.
 /// ```text
 /// variant {
 ///    controllers;
+///    allowed_viewers: vec principal;
 ///    public;
 /// }
 /// ```
-#[derive(Default, Clone, Copy, CandidType, Deserialize, Debug, PartialEq, Eq, EnumIter)]
+#[derive(Default, Clone, CandidType, Deserialize, Debug, PartialEq, Eq, EnumIter)]
 pub enum LogVisibility {
     #[default]
     #[serde(rename = "controllers")]
-    Controllers = 1,
+    Controllers,
+    #[serde(rename = "allowed_viewers")]
+    AllowedViewers(BoundedAllowedViewers),
     #[serde(rename = "public")]
-    Public = 2,
+    Public,
 }
 
+// TODO(EXC-1670): remove after migration to `pb_canister_state_bits::LogVisibilityV2`.
 impl From<&LogVisibility> for pb_canister_state_bits::LogVisibility {
     fn from(item: &LogVisibility) -> Self {
+        use pb_canister_state_bits as pb;
         match item {
-            LogVisibility::Controllers => pb_canister_state_bits::LogVisibility::Controllers,
-            LogVisibility::Public => pb_canister_state_bits::LogVisibility::Public,
+            LogVisibility::Controllers => pb::LogVisibility::Controllers,
+            LogVisibility::AllowedViewers(_) => pb::LogVisibility::Controllers,
```

Comment:
We have a problem here. Initially management canister enum `LogVisibility` was a rust's `unit-only enum` with all the variants able to be converted into `i32`. After adding `AllowedViewers(vec)` variant this is not longer the case.

So first we have to remove explicit numbers. We even may change the order, but only for this rust type, not for the proto type.

In order to keep round-trip serialization a no-op I can add a dummy `pb::LogVisibility::EmptyAllowedViewers` enum variant `= 3`. That will keep the round-trip serialisation *sort of* no-op, only if rust type came with empty list.

Would it be ok for this step?

On the follow-up step I plan to add a new `pb::LogVisibilityV2` that will be able to hold both the enum type and extra data, eg.

```proto

enum LogVisibilityEnum {
  LOG_VISIBILITY_ENUM_UNSPECIFIED = 0;
  LOG_VISIBILITY_ENUM_CONTROLLERS = 1;
  LOG_VISIBILITY_ENUM_ALLOWED_VIEWERS = 2;
  LOG_VISIBILITY_ENUM_PUBLIC = 3;
}

message LogVisibilityV2 {
  LogVisibilityEnum log_visibility_enum = 1;
  repeated types.v1.PrincipalId allowed_viewers = 2;
}
```

WDYT?

Reply:
UPD: 

After an off-line discussion with @schneiderstefan we decided to change the order of PRs and first to update protobuf types to support new enum variant with list of principles.

I am going to make this PR a draft for now.

Reply:
I was actually going to say that if indeed we need a V2 (and I get we need it now with a non-unit enum), I'd also recommend we first introduced the V2 in protobuf to add support for the new variant and then change the candid interface of the management canister (in fact, I think this might be to our advantage anyway as it gives us a clean way to introduce things in steps maintaining backward/forward compatibility).

Reply:
I was actually going to say that if indeed we need a V2 (and I get we need it now with a non-unit enum), I'd also recommend we first introduced the V2 in protobuf to add support for the new variant and then change the candid interface of the management canister (in fact, I think this might be to our advantage anyway as it gives us a clean way to introduce things in steps maintaining backward/forward compatibility).

Reply:
I think if we used a `oneof` it might actually be cleaner. Something like:
```
message LogVisibilityV2 {
  oneof log_visibility {
    int32 controllers = 1;
    repeated types.v1.PrincipalId allowed_viewers = 2;
    int32 public = 3;
  }
}
```

The benefit I see is that we have encoded all necessary information at the type level. With your approach we'd have to make sure that allowed_viewers is empty unless the enum is `LOG_VISIBILITY_ENUM_ALLOWED_VIEWERS` or ignore it in any other case, i.e. we'd need some extra code to enforce that nothing strange is going on.

Reply:
I have applied `oneof` suggestion as well as introduced v2 for ic00 types.

@dsarlis, @schneiderstefan PTAL.